datacube/api/core.py

```diff
@@ -430,6 +436,14 @@ def filter_jan(dataset): return dataset.time.begin.month == 1
             if extra_dims.has_empty_dim():
                 return xarray.Dataset()
 
+        if resolution is not None:
+            _LOG.warning("resolution is now expected in (x, y) order if provided as a tuple, "
```

Comment:
That's not quite a correct statement. in `odc.geo` there is no "default order" for x,y vs y,x. But there are helper methods to explicitly state what order you are supplying:

```python
from odc.geo import resxy_, resyx_, res_

resxy_(10, -10) == resyx_(-10, 10), res_(10) == resxy_(10, -10), res_(10) == resyx_(-10, 10)
``` 

But really supplying a single float/int is a preferred way for default "flipped y-axis, square pixels".

Reply:
Yes, but in the interest of removing ambiguity around when x,y vs y,x ordering is used I thought it might make sense to expect resolution in x,y if not being provided as a single float/int. But I'm happy to change the warning message if that's counterintuitive.

Reply:
Not sure about this code, but anywhere in `odc.geo` where resolution is expected, passing in just a tuple SHOULD raise an error, exactly because it's such a common mistake to make.

https://odc-geo.readthedocs.io/en/latest/migration.html#resolution

But yes using a single float value is a preferred way, so for code migration

- `resolution=(-X, X)` should be replace with just `X`
- `resolution=(A, B)`  should be replaced with `resyx_(A,B)`, or if it makes more sense in the context `resxy_(B,A)`.

Reply:
In this context, does it make more sense to expect x,y or y,x ordering if the resolution does need to be provided as a tuple?

Reply:
It should not be provided as tuple, providing just a tuple is an ERROR, exactly because it's not well defined what order is implied.

Reply:
original order for resolution in a tuple was `(y, x)`, now you can either

- Support old style approach by assuming that previous order `(y,x)` is supplied and so feed it through `resyx_` before passing it on to `odc.geo`.
- Adopt `odc.geo` approach of raising an error when that happens

Reply:
Right, gotcha. Raising warnings instead of errors to keep backwards compatibility for now, but hopefully the warning message is better now